src/kOS.Safe/Encapsulation/StringValue.cs

```diff
@@ -342,6 +342,12 @@ string IConvertible.ToString(IFormatProvider provider)
 
         object IConvertible.ToType(Type conversionType, IFormatProvider provider)
         {
+            if (conversionType == typeof(StringValue))
+                return this;
+            else if (conversionType == typeof(BooleanValue))
+                return new BooleanValue(string.IsNullOrEmpty(internalString) ? false : true);
```

Comment:
I'm not entirely sure about treating an existent but empty string as false.  It's different from a null reference to a string, which is traditionally the only thing that is interpreted as false this way.  But since kerboscript doesn't let people do anything with nulls anyway, I guess this will do.